hanabi/game/static/game/scripts/game-in-session.js

```diff
@@ -124,16 +98,79 @@ function populateDisplay( gameState ) {
 
     // Display deck.
     let deckImg = document.createElement("img");
+    deckImg.id = ID_DECK_IMG;
     deckImg.classList.add("card");
-    deckImg.src = cardBackLink;
+    deckImg.src = LINK_CARD_BACK;
     $("#deck").append(deckImg);
     let deckCount = document.createElement("h3");
+    deckCount.id = ID_CARDS_REMAINING;
     deckCount.innerHTML = "Remaining:</br>" + remainingCards;
     deckCount.classList.add("centered");
     $("#deck").append(deckCount);
 
     // Display Bomb Count.
     let bombCountTitle = document.createElement("h3");
+    bombCountTitle.id = ID_BOMBS;
     bombCountTitle.innerText = "Bombs remaining: " + remainingBombs;
     $("#bombs").append(bombCountTitle);
+
+    PENULTIMATE_GAME_STATE = gameState;
+}
+let ID_DECK_IMG = 'deck-image';
+let ID_BOMBS = 'bombs-remaining';
+let ID_CARDS_REMAINING = 'cards-remaining';
+let CARD_ID_SUFFIX = "scards"; // playerName + suffix
+
+let LINK_BASE_CARD = "/static/hanabi_deck/";
+let LINK_CARD_BACK = LINK_BASE_CARD + "card_back.png";
+
+let PENULTIMATE_GAME_STATE = {};
+
+function updateDisplay(gameState) {
+    console.log(gameState);
+    let remainingBombs = gameState["remaining_bombs"];
+    let remainingCards = gameState["remaining_cards"];
+    let players = gameState["players"];
+    console.log('updating');
+    let bombCount = document.getElementById(ID_BOMBS);
+    let deckCount = document.getElementById(ID_CARDS_REMAINING);
+
+    if (bombCount.innerText !== "Bombs remaining: " + remainingBombs) {
+        bombCount.innerText = "Bombs remaining: " + remainingBombs;
+    }
+    if (deckCount.innerHTML !== "Remaining:</br>" + remainingCards) {
+        deckCount.innerHTML = "Remaining:</br>" + remainingCards;
+    }
```

Comment:
It should be faster to just set the inner text to what you want rather than doing a check and set.